workflow/controller/operator.go

```diff
@@ -1704,14 +1704,12 @@ func hasOutputResultRef(name string, parentTmpl *wfv1.Template) bool {
 }
 
 // getStepOrDAGTaskName will extract the node from NodeStatus Name
-func getStepOrDAGTaskName(nodeName string, hasRetryStrategy bool) string {
+func getStepOrDAGTaskName(nodeName string) string {
 	if strings.Contains(nodeName, ".") {
 		name := nodeName[strings.LastIndex(nodeName, ".")+1:]
-		// Check retry scenario
```

Comment:
Not only `retry` scenario has `(`, but also `withItems` and `withParam`. Simply removed the `retry` condition check, and do it for all - This is based on the fact that neither template name nor step/DAG name is allowed to contain `(`.